### PR TITLE
Hot-reload work check

### LINKED ISSUE #107: [1pt] Verify that cartridge hot-reload works
Verify (and fix if it's not so) cartridge roles (and crud as separate module) supports hot-reload.

See also https://github.com/tarantool/cartridge/pull/1105

### PR DESCRIPTION
**Pull request by mRrvz**
Added hot-reload health test. 
Added ``stop()`` functions for ``crud`` roles.
Closes #107 #148

### CODE REVIEW

**Comment by rosik** [comment]
I'd suggest cleaning up globals `_G.crud` and `_G._crud`:

https://github.com/tarantool/crud/blob/c961e166645aeca897895f8513cc71e6a69c408c/crud.lua#L79-L96

You can use cartridge `stop` callback. See for example https://github.com/tarantool/cartridge/blob/master/cartridge/roles/vshard-storage.lua#L48

**Comment by no1seman** [comment]
There is another obe case that must be checked: during roles reload version of crud (and also all lua-files) may be changed on disk. Test must check, that new version fully reloaded.

File: crud.lua [hunk 1]
```diff
@@ -95,4 +95,13 @@ function crud.init_router()
    rawset(_G, 'crud', crud)
 end
 
+
+function crud.router_stop()
```
Comment (dokshina) [inline crud.lua:99]: But you use `crud.stop_router()` above/

File: test/integration/reload_test.lua [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    end)
+
+    local cnt = #g.insertions_passed
+    local ok = pcall(g.cluster.retrying, g.cluster, {timeout = 2}, function()
+        helpers.assert_ge(#g.insertions_passed, cnt+1)
+    end)
+
+    if not ok then
+        g.highload_fiber:cancel()
+        fiber.sleep(10000)
```
Comment (dokshina) [inline test/integration/reload_test.lua:133]: ???

**Review by dokshina — APPROVED** [review]

**Pull request merged**